qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -858,31 +905,51 @@ def status(self) -> str:
         ):
             return "INITIALIZING"
 
+        # Backend jobs
         statuses = set()
         with self._job_futures.lock:
-            for idx, item in enumerate(self._job_futures):
-                kwargs, fut = item
-                job = kwargs["job"]
-                job_status = job.status()
-                statuses.add(job_status)
-                if job_status == JobStatus.ERROR:
-                    job_err = "."
-                    if hasattr(job, "error_message"):
-                        job_err = ": " + job.error_message()
-                    self._errors.append(f"Job {job.job_id()} failed{job_err}")
+            for idx, (kwargs, fut) in enumerate(self._job_futures):
+                for job in kwargs["jobs"]:
+                    job_status = job.status()
+                    statuses.add(job_status)
+                    if job_status == JobStatus.ERROR:
+                        job_err = "."
+                        if hasattr(job, "error_message"):
+                            job_err = ": " + job.error_message()
+                        self._errors.append(f"Job {job.job_id()} failed{job_err}")
 
                 if fut.done():
                     self._job_futures[idx] = None
                     ex = fut.exception()
                     if ex:
+                        jobs = [job.job_id() for job in kwargs["jobs"]]
                         self._errors.append(
-                            f"Post processing for job {job.job_id()} failed: \n"
+                            f"Post processing for job {jobs} failed: \n"
                             + "".join(traceback.format_exception(type(ex), ex, ex.__traceback__))
                         )
                         statuses.add(JobStatus.ERROR)
-
             self._job_futures = ThreadSafeList(list(filter(None, self._job_futures)))
 
+        # Analysis jobs
+        proc_statuses = set()
+        with self._analysis_futures.lock:
+            for idx, (kwargs, fut) in enumerate(self._analysis_futures):
+                if fut.done():
+                    self._analysis_futures[idx] = None
+                    ex = fut.exception()
+                    if ex:
+                        callback = kwargs["callback"]
+                        self._errors.append(
+                            f"Post processing for callback {callback} failed: \n"
+                            + "".join(traceback.format_exception(type(ex), ex, ex.__traceback__))
+                        )
+                        proc_statuses.add(JobStatus.ERROR)
+
+            self._analysis_futures = ThreadSafeList(list(filter(None, self._analysis_futures)))
+
+        if JobStatus.ERROR in proc_statuses:
+            return JobStatus.ERROR.name
```

Comment:
The current experiment status is mapped in the order specified in the docstring. So if a job is in `INITIALIZING`, the experiment status is considered to be in `INITIALIZING`, even if a previous job has failed. If we wan to make `ERROR` as dominant status (e.g. experiment is in ERROR if any failed), then we should change both the docstring and the order in like 953-960.